cli/src/commands/init.rs

```diff
@@ -33,9 +33,11 @@ pub(crate) struct InitArgs {
     /// The destination directory
     #[arg(default_value = ".", value_hint = clap::ValueHint::DirPath)]
     destination: String,
+    /// DEPRECATED: Use `jj git init`
     /// Use the Git backend, creating a jj repo backed by a Git repo
     #[arg(long)]
```

Comment:
Since this is now deprecated, we should add `hide = true`, or what do you think? Same for `git_repo` below.

Reply:
Nit: I'd hide it and then print a warning message if we encounter it on the commandline, see #2747.

Reply:
Good point. That would also match what we're doing for `jj checkout` and `jj merge`.

Reply:
Thanks. I missed that earlier. I've hidden both flags now.